libicyque.c

```diff
@@ -1225,6 +1350,75 @@ icq_fetch_events_cb(IcyQueAccount *ia, JsonObject *obj, gpointer user_data)
 	icq_fetch_url_with_method(ia, "GET", fetchBaseURL, NULL, icq_fetch_events_cb, NULL);
 }
 
+static void
+icq_robusto_add_client_cb(IcyQueAccount *ia, JsonObject *obj, gpointer user_data)
+{
+	JsonObject *status = json_object_get_object_member(obj, "status");
+	if(json_object_get_int_member(status, "code") != 20000) {
+		purple_connection_error(ia->pc, PURPLE_CONNECTION_ERROR_AUTHENTICATION_FAILED, "Failed to register client at robusto API.");
+	} else {
+		JsonObject *results = json_object_get_object_member(obj, "results");
+		guint64 clientId = json_object_get_int_member(results, "clientId");
+		ia->robusto_client_id = clientId;
+		
+		purple_connection_set_state(ia->pc, PURPLE_CONNECTION_CONNECTED);
+		purple_debug_info("icyque", "Authentication succeeded. Starting fetch-loop.\n");
+		icq_fetch_url_with_method(ia, "GET", ia->last_fetchBaseURL, NULL, icq_fetch_events_cb, NULL);
+	}
+}
+
+static void icq_robusto_add_client(IcyQueAccount *ia)
+{
+	JsonObject *addClientParams = json_object_new();
+	JsonObject *addClientParamsUserAgent = json_object_new();
+	json_object_set_object_member(addClientParams, "ua", addClientParamsUserAgent);
+	json_object_set_string_member(addClientParamsUserAgent, "app", "icq"); //TODO pretend to be official client or Pidgin?
+	json_object_set_string_member(addClientParamsUserAgent, "build", "1");
+	json_object_set_string_member(addClientParamsUserAgent, "label", "webicq");
+	json_object_set_string_member(addClientParamsUserAgent, "os", "win");
+	json_object_set_string_member(addClientParamsUserAgent, "version", "0.1");
+	
+	JsonObject *addClientRequest = icq_generate_robusto_request(ia, "addClient", addClientParams);
+
+	const gchar* addClientRequestStr = json_object_to_string(addClientRequest);

```

Comment:
`json_object_to_string` returns a non-const char* that needs to be `g_free`d after using to prevent a memleak